lib/recurly/pricing/checkout/index.js

```diff
@@ -238,47 +234,47 @@ export default class CheckoutPricing extends Pricing {
         this.items.adjustments.push(adjustment);
       }
 
-      debug('set.adjustment');
-      this.emit('set.adjustment', adjustment);
-      resolve(adjustment);
+      this.resolveAndEmit('set.adjustment', adjustment, resolve);
     }, this);
   }
 
   /**
    * Updates coupon. Manages a single coupon on the item set, unsetting any exisitng
    * coupon
    *
+   * `set.coupon` is emitted when a valid coupon is set
+   * `unset.coupon` is emitted when an existing valid coupon is removed
+   * `error.coupon` is emitted when the requested coupon is invalid
+   *
    * @param {String} couponCode
    * @return {PricingPromise}
    * @public
    */
   coupon (couponCode) {
-    const unset = () => {
-      debug('unset.coupon');
-      delete this.items.coupon;
-      this.emit('unset.coupon');
-    };
+    if (~this.couponCodes.indexOf(couponCode)) {
```

Comment:
Curious: what is this doing? Could we assign this to a new variable that describes what this does?

Reply:
It's a common way of detecting if an element is in an Array